### PR TITLE
DATACOUCH-295 Add documentation for reactive support

### PR DESCRIPTION
**Pull request by subalakr**

### CODE REVIEW

File: src/main/asciidoc/reactiverepository.adoc [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+[[couchbase.reactiverepository.intro]]
+== Introduction
+
+This chapter describes the reactive repository support for couchbase. This builds on the core repository support explained in <<couchbase.repository>>.
+So make sure you’ve got a sound understanding of the basic concepts explained there.
+
+[[couchbase.reactiverepository.libraries]]
+== Reactive Composition Libraries
+
+Couchbase Java SDK 2.0 has taken a reactive programming approach since its inception using an early reactive extension to JVM, https://github.com/ReactiveX/RxJava/tree/1.x/[RxJava1].
```
Comment (daschl) [inline src/main/asciidoc/reactiverepository.adoc]: "SDK 2.x" ?

File: src/main/asciidoc/reactiverepository.adoc [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+It provides RxJava1 observable sequence API to compose asynchronous database operations.
+
+Reactive Couchbase repositories provide project Reactor wrapper types and can be used by simply extending from one of the library-specific repository interfaces:
+
+ReactiveCrudRepository
```
Comment (daschl) [inline src/main/asciidoc/reactiverepository.adoc]: should this be a list?

File: src/main/asciidoc/reactiverepository.adoc [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+ReactiveSortingRepository
+
+Spring-data-couchbase converts RxJava 1 observables to reactor types by using reactive-streams adapters from https://github.com/ReactiveX/[RxJavaReactiveStreams]
+for convenience since these conversions can easily clutter application code. This transformation happens on same thread. It also provides direct access to RxJava1 observable
+sequence API from SDK through RxJavaCouchbaseOperations methods.
+
+[[couchbase.reactiverepository.usage]]
+== Usage
+To access domain entities stored in a Couchbase you can leverage our sophisticated repository support that eases implementing those quite significantly.
```
Comment (daschl) [inline src/main/asciidoc/reactiverepository.adoc]: "In a couchbase .. bucket/database ?"

File: src/main/asciidoc/reactiverepository.adoc [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+[source,java]
+----
+public class PersonRepositoryTests {
+
+    @Autowired
+    ReactivePersonRepository repository;
+
+    @Test
+    public void sortsElementsCorrectly() {
+      Flux<Person> persons = repository.findAll(Sort.by(new Order(ASC, "lastname")));
```
Comment (daschl) [inline src/main/asciidoc/reactiverepository.adoc]: should this test have at least a simple assert?

Comment (daschl) [inline src/main/asciidoc/reactiverepository.adoc]: (if someone copy pastes it)

File: src/main/asciidoc/reactiverepository.adoc [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    }
+}
+----
+====
+
+[[couchbase.reactiverepository.querying]]
+== Repositories and Querying
+
+Spring Data's Reactive Couchbase comes with full querying support already provided by the blocking <<couchbase.repository.querying>>.
```
Comment (daschl) [inline src/main/asciidoc/reactiverepository.adoc]: is the stopping w/o a further example intentional?

File: src/main/asciidoc/reactiverepository.adoc (hunk 4 above)
Comment (subalakr) [inline src/main/asciidoc/reactiverepository.adoc]: Fixed

**Comment by subalakr** [comment]
Merged.

**Pull request closed without merging**